android/app/src/main/java/com/sphereon/musaprn/MusapModule.kt

```diff
@@ -0,0 +1,27 @@
+package com.sphereon.musaprn;
+
+import com.facebook.react.bridge.ReactApplicationContext
+import com.facebook.react.bridge.ReactContextBaseJavaModule
+import com.facebook.react.bridge.ReactMethod
+import com.fasterxml.jackson.module.kotlin.jacksonObjectMapper
+import fi.methics.musap.sdk.api.MusapClient
+
+private val objectMapper = jacksonObjectMapper()
+
+
+class MusapModule(val context: ReactApplicationContext) : ReactContextBaseJavaModule(context) {
+
+    override fun getName(): String = "MusapModule"
+
+    @ReactMethod(isBlockingSynchronousMethod = true)
+    fun listEnabledSscdsAsJson(): String {
```

Comment:
This is going to be a nightmare on the typescript side if you encode everything to string. That means every implementation has to do the work this module should be doing

Reply:
I totally agree, however we cannot serialize everything because Android uses Gson for serialization/deserialization and it's not possible to override the default configuration. There is an experimental way to do it, called [Turbo Modules](https://github.com/reactwg/react-native-new-architecture/blob/main/docs/turbo-modules.md), it uses C++ to call the native methods and there is no JSON involved at all. I tried it last weekend but I couldn't get it working with MUSAP.